File: C/exo2_1.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>

int main()
{
    pid_t pid_fils=fork();
    if(pid_fils==0)
        printf("Je suis le fils.\n");
    else if(pid_fils>0)
    {
        wait(NULL);
        printf("Je suis le père.\n");
    }
    else
        perror("Erreur lors de la création du processus fils.\n");

    return 0;
}

File: python/exo2_1.py
import os
import sys

pid = os.fork()

if pid == 0:
    print("Je suis le fils.")
    sys.exit(0)
elif pid > 0:
    os.wait()
    print("Je suis le père.")
else:
    print("Erreur lors de la création du processus fils.", file=sys.stderr)